react-accordion/my-app/src/Accordion.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from 'react';
+
+interface Topic {
+  id: number;
+  title: string;
+  content: string;
+}
+
+type AccordionProps = {
+  topics: Topic[];
+};
+
+export function Accordion({ topics }: AccordionProps) {
+  // takes in an array ot topic objects as its Props
+  const [visibleTopicId, setVisibleTopicId] = useState<number | null>(null);
```

Comment:
I recommend that you very, very rarely use `null`. It's almost always easier and cleaner to use `undefined`, instead. Thanks to how `useState` is defined, this is very simple with:
`useState<number>();`

and then you can clear it with `setVisibleTopicId()`.